src/Core/Resolvers/IQueryBuilder.cs

```diff
@@ -64,6 +64,15 @@ public interface IQueryBuilder
         /// <returns></returns>
         public string BuildStoredProcedureResultDetailsQuery(string databaseObjectName);
 
+        /// <summary>
+        /// Returns the query to get the read-only columns present in the table.
+        /// </summary>
+        /// <param name="schemaOrDatabaseParamName">Param name of the schema/database.</param>
+        /// <param name="tableParamName">Param name of the table.</param>
+        /// <exception cref="NotImplementedException">Thrown when class implementing the interface has not provided
+        /// an overridden implementation of the method.</exception>
+        string BuildQueryToGetReadOnlyColumns(string schemaOrDatabaseParamName, string tableParamName);
```

Comment:
I just stuck to the placements that we have now. Even for the queries that collect metadata, they are placed in {*}QueryBuilder classes. And yeah I think QueryBuilder is still a better option, since it deals with constructing the query. MetadataProviders should be providing metadata and should not have to deal with query generation/construction.